go/vt/vtgate/planbuilder/testdata/memory_sort_cases.json

```diff
@@ -606,49 +606,58 @@
       "QueryType": "SELECT",
       "Original": "select user.col1 as a, user.col2 b, music.col3 c from user, music where user.id = music.id and user.id = 1 order by c",
       "Instructions": {
-        "OperatorType": "Sort",
-        "Variant": "Memory",
-        "OrderBy": "(2|3) ASC",
-        "ResultColumns": 3,
+        "OperatorType": "SimpleProjection",
+        "Columns": [
+          0,
+          1,
+          2
+        ],
         "Inputs": [
           {
-            "OperatorType": "Join",
-            "Variant": "Join",
-            "JoinColumnIndexes": "L:1,L:2,R:0,R:1",
-            "JoinVars": {
-              "user_id": 0
-            },
-            "TableName": "`user`_music",
+            "OperatorType": "Sort",
+            "Variant": "Memory",
```

Comment:
this looks like a slower plan

Reply:
https://github.com/vitessio/vitess/pull/12954#issuecomment-1520004048